app.rb

```diff
@@ -0,0 +1,84 @@
+require 'sinatra/base'
+require 'sinatra/flash'
+require 'uri'
+require 'sinatra/reloader'
+require './lib/peep'
+require './lib/user'
+require './database_connection_setup'
+
+class Chitter < Sinatra::Base
+  enable :sessions
+  register Sinatra::Flash
+
+  configure :development do
+    register Sinatra::Reloader
+  end
+
+  get '/' do
+    'Peeps'
+  end
+
+  get '/peeps' do
+    @user = User.find(user_id: session[:user_id])
+    @peeps = Peep.all
+    erb :'peeps/index'
+  end
+
+  get '/peeps/new' do
+    @user = User.find(user_id: session[:user_id])
+    erb :'peeps/new'
+  end
+
+  post '/peeps' do
+    Peep.create(peep_text: params[:message], user_id: session[:user_id])
+    redirect '/peeps'
+  end
+
+  get '/users/new' do
+    erb :'users/new'
+  end
+
+  post '/users' do
+    # This if statement never actually runs in live, only for capybara feature tests - see test.
+    if params.values.all? { |x| x.empty? }
+      flash[:notice] = "Please check the required fields."
+      redirect '/users/new'
+    end
+
+    user = User.create(email: params[:email], password: params[:password], name: params[:name], username: params[:username])
+    
+    if user.nil?
+      flash[:notice] = "An account with that email already exists."
+      redirect('/users/new')
+    elsif user == false
```

Comment:
I like this alternative messages based off whether the user is left blank, or duplicates an existing